libcaf_core/caf/json_builder.test.cpp

```diff
@@ -93,32 +95,170 @@ TEST("empty JSON value") {
   check(val.is_null());
 }
 
-TEST("integer") {
+TEST("int8") {
+  check(builder.value(int8_t{42}));
+  auto val = builder.seal();
+  check(val.is_integer());
+  check_eq(val.to_integer(), 42);
+}
+
+TEST("uint8") {
+  check(builder.value(uint8_t{42}));
+  auto val = builder.seal();
+  check(val.is_integer());
+  check_eq(val.to_integer(), 42);
+}
+
+TEST("int16") {
+  check(builder.value(int16_t{42}));
+  auto val = builder.seal();
+  check(val.is_integer());
+  check_eq(val.to_integer(), 42);
+}
```

Comment:
There's quite a bit of repetition here that we can cut. The only real difference is the call to `value`:

```c++
TEST("integer") {
  SECTION("int8") {
    check(builder.value(int8_t{42}));
  }
  // ... sections for the other integer types ...
  auto val = builder.seal();
  check(val.is_integer());
  check_eq(val.to_integer(), 42);
}